src/app/src/main/java/com/example/bookmark/util/RequestUtil.java

```diff
@@ -28,6 +28,7 @@ public static void retrieveRequestsOnBookByStatus(
                 for (Request r: requestList) {
                     if (r.getStatus().equals(status)) {
                         onSuccessListener.onSuccess(r);
+                        return;
                     }
                 }
```

Comment:
Just adding this in here in case no books match:
```suggestion
                }
                onSuccessListener.onSuccess(null);
```